src/arithmetic/algebraic_expression/algebraic_expression_operand.c

```
@@ -67,6 +67,58 @@ const AlgebraicExpression *AlgebraicExpression_DestOperand
+	// create local queue for DFS style traversal
+	// visiting nodes from left to right
+	// NOTE: this function doesn't treat operands switch by transpose operation
+	// e.g. T(A*B) -> Bt * At
+	// will treat A as the leftmost operand and B as the rightmost
+	AlgebraicExpression **queue = array_new(AlgebraicExpression*, 1);
+	array_append(queue, (AlgebraicExpression*)root);
+
+	// as long as there are nodes to visit
+	while(array_len(queue) > 0) {
+		// get the newest node added to the queue
+		AlgebraicExpression *current = array_pop(queue);
```

Cppcheck (2.10-2)

[error] Common realloc mistake

(memleakOnRealloc)